sentry_streams/sentry_streams/adapters/arroyo/reduce.py

```diff
@@ -66,6 +75,64 @@ def accumulator(self, result: Any, value: BaseValue[RoutedValue]) -> RoutedValue
         return routed
 
 
+class MetricsAccumulator(Accumulator[Message[Any], Any]):
```

Comment:
Also why does this extend `Accumulator[Message[Any, Any]] rather than being generic itself? It seems we are loosing type safety here as the accumulator is expected to be `Accumulator[Any, Any]` and we cast it as `Accumulator[PipelineMessage[InputType], OutputType]]` below. This could be enforced statically. 
Was there any issue you could not address ?